src/proxyfactory.rs

```diff
@@ -80,6 +80,25 @@ impl ProxyFactory {
         let mut result: ProxyResult = Default::default();
         let drain = proxy_drain.unwrap_or_else(|| self.drain.clone());
 
+        let mut resolver = None;
+        // Optionally create the DNS proxy.
+        if self.config.dns_proxy {
```

Comment:
We probably need to redo this config flag logic at some point - with this you can't really provide a resolver without enabling the whole DNS proxy.